### PR TITLE
Pointers instead of copies

### PR DESCRIPTION
**Pull request by rogeruiz**
pairing with @LinuxBozo

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func pipe(r io.Reader, w io.WriteCloser) {
 }
 
 // pass a stopChan to the go routine
-func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan *chan struct{}) {
+func proxy(local *net.TCPConn, remoteAddr *net.TCPAddr, stopChan *chan struct{}) {
```
Comment (jmcarp) [inline main.go]: It doesn't look like this changes the behavior of this function--`*.net.TCPConn` implements `io.ReadWriteCloser`. What is this for?

File: main.go [hunk 2]
```diff
@@ -76,8 +76,8 @@ func pipe(r io.Reader, w io.WriteCloser) {
 	fmt.Println("Closing pipe")
 }
 
-// pass a stopChan to the go routtine
-func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan chan struct{}) {
+// pass a stopChan to the go routine
+func proxy(local io.ReadWriteCloser, remoteAddr *net.TCPAddr, stopChan *chan struct{}) {
```
Comment (jmcarp) [inline main.go]: How does this change the behavior of this function? We aren't mutating the channel here, so what changes when we pass a pointer?

Comment (rogeruiz) [reply main.go]: My understanding is that we're mutating the state of it in the `master` function. but I will say that I'm only beginning to understand simple pointers. I think I fixed the proxy in 78d72bbd4d0d4a3f65edb406453c2aada45c5e10

**Comment by rogeruiz** [comment]
whoops, posted in the wrong thread 18F/kubernetes-broker#57 👀

**Comment by jmcarp** [comment]
It's not clear what problem this patch is solving, and I'm not sure that all the changes in here are relevant or necessary. If the real fix lives in https://github.com/18F/redis-sentinel-proxy/commit/78d72bbd4d0d4a3f65edb406453c2aada45c5e10, how about a clean PR with just that change?

**Comment by rogeruiz** [comment]
Closing #2 in light of #3, thanks @cnelson

**Pull request closed without merging**